src/main/java/bio/terra/common/iam/SamUserFactory.java

```diff
@@ -59,6 +63,13 @@ public SamUser from(BearerToken bearerToken, String samBasePath) {
   @VisibleForTesting
   UsersApi createUsersApi(BearerToken bearerToken, String samBasePath) {
     ApiClient samApiClient = new ApiClient();
+    OkHttpClient okHttpClient = samApiClient
```

Comment:
The interceptor piece LGTM, but I have a concern with this pattern of creating a new ApiClient (with a new underlying OkHttpClient) object for each request. `OkHttpClient`s are heavyweight objects which maintain their own threadpools, and the documented recommendation is to share one across requests (https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/). This has been a significant performance issue for WSM in the past